src/components/componentContainer.ts

```diff
@@ -21,10 +21,10 @@ export class ComponentContainer extends Component {
     this._detachCallback = (component: Component) => this.remove(component);
   }
 
-  public anchor(selection: SimpleSelection<void>) {
+  public anchor(selection: d3.Selection<HTMLElement, any, any, any>) {
```

Comment:
The `_element` and the `_rootElement` use this as well; agreed this is janky. In general we need to revisit the Selection types everywhere and I don't want to band-aid over the situation more than it already is (in which we use SimpleSelection in most places which throws all typing away)